From ab9310eb8843d44873343ae2749f35c66c9e23ce Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 29 Mar 2023 18:01:49 +0300
Subject: [PATCH 1/1] compulab: imx9: Update ucm-imx93 files

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/clk/imx/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/imx/Makefile b/drivers/clk/imx/Makefile
index 5bba478a50..0fb7459405 100644
--- a/drivers/clk/imx/Makefile
+++ b/drivers/clk/imx/Makefile
@@ -20,4 +20,5 @@ obj-$(CONFIG_$(SPL_TPL_)CLK_IMX8MP) += clk-imx8mp.o clk-pll14xx.o \
 
 obj-$(CONFIG_$(SPL_TPL_)CLK_IMXRT1020) += clk-imxrt1020.o
 obj-$(CONFIG_$(SPL_TPL_)CLK_IMXRT1050) += clk-imxrt1050.o
+obj-$(CONFIG_$(SPL_TPL_)CLK_IMX93) += clk-imx93.o clk-fracn-gppll.o clk-gate-93.o
 obj-$(CONFIG_CLK_IMX8MP) += clk-hdmimix.o
-- 
2.17.1

