From 065675a5417a9779e68916265db72880a3f3b0e5 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 7 Apr 2023 10:40:51 +0300
Subject: [PATCH] drivers: phy: Fix mergre issue

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/phy/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/phy/Kconfig b/drivers/phy/Kconfig
index 6edfa98486..cdbf5a96c9 100644
--- a/drivers/phy/Kconfig
+++ b/drivers/phy/Kconfig
@@ -281,6 +281,15 @@ config PHY_IMX8MQ_USB
 	help
 	  Support the USB3.0 PHY in NXP i.MX8MQ SoC
 
+config PHY_IMX93_MIPI_DPHY
+	bool "NXP i.MX93 MIPI DPHY Driver"
+	depends on PHY
+	depends on ARCH_IMX9
+	select REGMAP
+	help
+	  Enable this to add support for the Synopsys DW MIPI DPHY as found
+	  on NXP's i.MX93 SoC.
+
 config PHY_XILINX_ZYNQMP
 	tristate "Xilinx ZynqMP PHY driver"
 	depends on PHY && ARCH_ZYNQMP
-- 
2.17.1

