From 6bb8c83b6d5d5bd52ad42b436559ca6d365c07c6 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 7 Apr 2023 10:43:29 +0300
Subject: [PATCH] drivers: video: startek-ili9881c: Add .plat_auto .data

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/video/startek-ili9881c.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/video/startek-ili9881c.c b/drivers/video/startek-ili9881c.c
index 6fe5581834..cb2eb6ac24 100644
--- a/drivers/video/startek-ili9881c.c
+++ b/drivers/video/startek-ili9881c.c
@@ -497,8 +497,19 @@ static const struct panel_ops ili9881c_panel_ops = {
 	.get_display_timing = ili9881c_panel_get_display_timing,
 };
 
+struct ili_platform_data {
+	int (*enable)(struct udevice *dev);
+};
+
+static int ili_enable(struct udevice *dev) {
+	return 0;
+}
+
+static const struct ili_platform_data ili_data = {
+	.enable = &ili_enable,
+};
 static const struct udevice_id ili9881c_panel_ids[] = {
-	{ .compatible = "startek,ili9881c" },
+	{ .compatible = "startek,ili9881c", .data = (ulong)&ili_data },
 	{ }
 };
 
@@ -509,5 +520,6 @@ U_BOOT_DRIVER(ili9881c_panel) = {
 	.ops			  = &ili9881c_panel_ops,
 	.probe			  = ili9881c_panel_probe,
 	.remove			  = ili9881c_panel_disable,
+	.plat_auto		= sizeof(struct mipi_dsi_panel_plat),
 	.priv_auto		= sizeof(struct ili9881c_panel_priv),
 };
-- 
2.17.1

