From 936aeb911228886c8ffcb9b3a7b275f6f49aa0da Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Sun, 7 May 2023 09:37:32 +0300
Subject: [PATCH] ucm-imx8m-plus: config: env: Add overlay load logics

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 include/configs/compulab-imx8m-plus.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/configs/compulab-imx8m-plus.h b/include/configs/compulab-imx8m-plus.h
index 2c8dbf3eae..9b58dcd911 100644
--- a/include/configs/compulab-imx8m-plus.h
+++ b/include/configs/compulab-imx8m-plus.h
@@ -83,6 +83,7 @@
 	"splashimage=0x50000000\0" \
 	"console=ttymxc1,115200 console=tty1\0" \
 	"fdt_addr_r=0x43000000\0"			\
+	"fdto_addr_r=0x43800000\0"			\
 	"fdt_addr=0x43000000\0"			\
 	"boot_fdt=try\0" \
 	"fdt_high=0xffffffffffffffff\0"		\
@@ -145,7 +146,14 @@
 		"ulbootscript=load ${iface} ${dev}:${part} ${loadaddr} ${script};\0" \
 		"ulimage=load ${iface} ${dev}:${part} ${loadaddr} ${image}\0" \
 		"ulfdt=if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-			"load ${iface} ${dev}:${part} ${fdt_addr_r} ${fdtfile}; fi;\0" \
+			"load ${iface} ${dev}:${part} ${fdt_addr_r} ${fdtfile}; " \
+				"if itest.s x != x${fdtofile}; then" \
+				    "load ${iface} ${dev}:${part} ${fdto_addr_r} ${fdtofile};" \
+				    "fdt addr ${fdt_addr_r}; fdt resize 0x8000; fdt apply ${fdt_oaddr_r};" \
+				"else" \
+				    "true;" \
+				"fi;" \
+			"fi;\0" \
 		"bootlist=sd_ul usb_ul emmc_ul\0" \
 	"bsp_bootcmd=echo Running BSP bootcmd ...; " \
 		"for src in ${bootlist}; do " \
-- 
2.17.1

